subprojects/internal-integ-testing/src/main/groovy/org/gradle/test/fixtures/server/sftp/SFTPServer.groovy

```diff
@@ -105,7 +105,7 @@ class SFTPServer extends ServerWithExpectations implements RepositoryServer {
 
     public void restart() {
         stop(true)
-        before()
+        before(port)
```

Comment:
It's a valid concern, although a remote possibility given the restart happens quickly.  A more salient question is do we need to restart the server to disconnect the clients?  As it turns out, we can disconnect the sessions without restarting the server, so I've changed it to just do this instead.